File: archive/mobile/NoiseMonitor/app/src/main/java/com/example/noisemonitor/ui/HistoryAdapter.java
package com.example.noisemonitor.ui;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.noisemonitor.R;
import com.example.noisemonitor.api.HistoryEvent;

import java.util.List;

public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.HistoryViewHolder> {

    private final Context context;
    private final List<HistoryEvent> eventList;

    public HistoryAdapter(Context context, List<HistoryEvent> eventList) {
        this.context = context;
        this.eventList = eventList;
    }

    @NonNull
    @Override
    public HistoryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_history_event, parent, false);
        return new HistoryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull HistoryViewHolder holder, int position) {
        HistoryEvent event = eventList.get(position);

        holder.time.setText(event.getTime());
        holder.dbLevel.setText(String.format("%d dB", event.getDbLevel()));

        switch (event.getStatus()) {
            case CRITICAL:
                holder.statusText.setText("CRITICAL");
                holder.statusIndicator.setImageResource(R.drawable.ic_status_offline); // Red dot
                break;
            case WARNING:
                holder.statusText.setText("WARNING");
                holder.statusIndicator.setImageResource(R.drawable.ic_status_warning); // Yellow dot
                break;
            case NORMAL:
            default:
                holder.statusText.setText("NORMAL");
                holder.statusIndicator.setImageResource(R.drawable.ic_status_normal); // Gray dot
                break;
        }
    }

    @Override
    public int getItemCount() {
        return eventList.size();
    }

    public static class HistoryViewHolder extends RecyclerView.ViewHolder {
        TextView time, dbLevel, statusText;
        ImageView statusIndicator;

        public HistoryViewHolder(@NonNull View itemView) {
            super(itemView);
            time = itemView.findViewById(R.id.text_view_time);
            dbLevel = itemView.findViewById(R.id.text_view_db_level);
            statusText = itemView.findViewById(R.id.text_view_status);
            statusIndicator = itemView.findViewById(R.id.image_view_status_indicator);
        }
    }
}

File: mobile/NoiseMonitor/app/src/main/java/com/example/noisemonitor/HistoryAdapter.kt
package com.example.noisemonitor

import android.content.res.ColorStateList
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.noisemonitor.databinding.ItemHistoryEventBinding
import java.text.SimpleDateFormat
import java.util.Locale

class HistoryAdapter(private val items: List<HistoryEvent>) :
    RecyclerView.Adapter<HistoryAdapter.HistoryViewHolder>() {

    inner class HistoryViewHolder(val binding: ItemHistoryEventBinding) :
        RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HistoryViewHolder {
        val binding = ItemHistoryEventBinding.inflate(
            LayoutInflater.from(parent.context), parent, false
        )
        return HistoryViewHolder(binding)
    }

    override fun onBindViewHolder(holder: HistoryViewHolder, position: Int) {
        val item = items[position]
        val context = holder.itemView.context

        holder.binding.textViewTime.text = formatTimestamp(item.timestamp)
        holder.binding.textViewDbLevel.text = "${item.noiseLevel} dB"
        holder.binding.textViewStatus.text = item.type

        val colorRes = when(item.type) {
            "NORMAL" -> R.color.primary
            "WARNING" -> R.color.warning
            "CRITICAL" -> R.color.error
            else -> R.color.text_secondary
        }
        
        // Используем backgroundTintList, так как в xml это View с background
        holder.binding.statusIndicator.backgroundTintList = 
            ColorStateList.valueOf(context.getColor(colorRes))
    }

    private fun formatTimestamp(timestamp: String): String {
        return try {
            // Предполагаем, что сервер присылает "yyyy-MM-dd HH:mm:ss"
            val inputFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.getDefault())
            val outputFormat = SimpleDateFormat("HH:mm:ss dd.MM.yyyy", Locale.getDefault())
            val date = inputFormat.parse(timestamp)
            outputFormat.format(date!!)
        } catch (e: Exception) {
            timestamp // Если не удалось распарсить, возвращаем как есть
        }
    }

    override fun getItemCount(): Int = items.size
}
